controllers/authorizeService.js

```diff
@@ -0,0 +1,120 @@
+var AccountingService = require('../db/schemas/accountingService'),
+    config = require('../config'),
+    uuid = require('node-uuid');
+
+var authorizeService = (function () {
+
+    /**
+     * Generates am aìKey.
+     */
+    var generateApiKey = function (callback) {
+        var apiKey = uuid.v4();
+
+        return callback(apiKey);
+    };
+
+    /**
+     * Check if the remote cliente is the WStore; otherwise return an error.
+     */
+    var checkRemoteClient = function (req, callback) {
+        var storeHostname = config.appHost + ':' + config.endpoints.charging.port;
+        var reqIp = req.ip.replace(/^.*:/, ''); // Parse IPv4 embedded in IPv6
+        var remoteHostname = reqIp + ':' + req.connection.remotePort;
+
+        if (remoteHostname === storeHostname) {
```

Comment:
This check will always fail. Remote port is a free port in the machine at the time of making the request not the port where the Charging Backend is running...